### ISSUE TITLE
Add rita.js files (4) to cdnjs

### ISSUE THREAD

**Issue by dhowe** [dhowe/RiTa#197]
https://github.com/cdnjs/cdnjs

**Comment by dhowe** [comment]
Please figure out what is required to do this -- their documentation is a bit confusing -- and list it here (don't actually do it yet)

**Comment by CyrusSUEN** [comment]
To add to cdnjs:

1. fork [cdnjs repo](https://github.com/cdnjs/cdnjs/fork)
2. git clone forked cdnjs repo
3. add RiTaJS library to ```cdnjs/ajax/libs/rita/1.1.12``` in the repo
4. add ```package.json``` to ```cdnjs/ajax/libs/rita/```
5. commit only once per library update
6. create the first PR including [author] and [new] in the title

Things to note:
* Try to sync the forked repo as close as possible to the upstream before PR and use git rebase instead of git merge 
> Please use git pull --rebase instead of git pull, use git rebase upstream/master instead of git merge upstream/master, so that we can avoid of meaningless merging.

* Make sure that the lib passes ```$ npm test``` before PR https://github.com/cdnjs/cdnjs#run-npm-test-to-check-all-is-well

* Do not include version number in the file name

* [Enabling npm(recommended) or git auto update](https://github.com/cdnjs/cdnjs#enabling-npmrecommended-or-git-auto-update)

**Comment by dhowe** [comment]
do you feel comfortable doing this?

**Comment by CyrusSUEN** [comment]
> do you feel comfortable doing this?

OK! In fact I've done this already.
See the commit in my fork: https://github.com/CyrusSUEN/cdnjs/commit/e948643468539d0a504ffceea1c4330a55a56500

Just one problem:
It says the lib should pass ```$ npm test```. But when I do this in RiTaJS repo it doesn't pass.
I know it can pass ```$ gulp test``` though. 
```bash
Cyrus-Mac-Pro:RiTaJS cyrussuen$ npm test

> rita@1.1.12 test /Users/cyrussuen/RiTaJS
> gulp test-npm

[21:35:15] Using gulpfile ~/RiTaJS/gulpfile.js
[21:35:15] Starting 'clean'...
[21:35:15] Finished 'clean' after 45 ms
[21:35:15] Starting 'build-nolex'...
[21:35:15] Starting 'build-lex'...
[21:35:15] rita.js 147.59 kB
[21:35:15] Finished 'build-nolex' after 295 ms
[21:35:16] rita-full.js 1.86 MB
[21:35:16] Finished 'build-lex' after 789 ms
[21:35:16] Starting 'build'...
[21:35:16] Finished 'build' after 12 μs
[21:35:16] Starting 'test-npm'...
[21:35:16] Starting 'test-only'...
[INFO] Source: lib/rita.js
[21:35:18] Finished 'test-npm' after 2.77 s
{ [Error: Cannot find module '/Users/cyrussuen/RiTaJS/lib/rita']
  message: 'Cannot find module \'/Users/cyrussuen/RiTaJS/lib/rita\'' }
{ files: 0,
  assertions: 0,
  failed: 0,
  passed: 0,
  runtime: 0,
  tests: 0,
  coverage: 
   { files: 0,
     statements: { covered: 0, total: 0 },
     branches: { covered: 0, total: 0 },
     functions: { covered: 0, total: 0 },
     lines: { covered: 0, total: 0 } } }
[21:35:19] Finished 'test-only' after 3.42 s
```

**Comment by dhowe** [comment]
please include the output from both 'npm test' and 'gulp test-npm'

**Comment by dhowe** [comment]
please include the output from both 'npm test' and 'gulp test-npm'

**Comment by CyrusSUEN** [comment]
> include the output from both 'npm test' and 'gulp test-npm'

```bash
Cyrus-Mac-Pro:RiTaJS cyrussuen$ npm test

> rita@1.1.12 test /Users/cyrussuen/RiTaJS
> gulp test-npm

[21:35:15] Using gulpfile ~/RiTaJS/gulpfile.js
[21:35:15] Starting 'clean'...
[21:35:15] Finished 'clean' after 45 ms
[21:35:15] Starting 'build-nolex'...
[21:35:15] Starting 'build-lex'...
[21:35:15] rita.js 147.59 kB
[21:35:15] Finished 'build-nolex' after 295 ms
[21:35:16] rita-full.js 1.86 MB
[21:35:16] Finished 'build-lex' after 789 ms
[21:35:16] Starting 'build'...
[21:35:16] Finished 'build' after 12 μs
[21:35:16] Starting 'test-npm'...
[21:35:16] Starting 'test-only'...
[INFO] Source: lib/rita.js
[21:35:18] Finished 'test-npm' after 2.77 s
{ [Error: Cannot find module '/Users/cyrussuen/RiTaJS/lib/rita']
  message: 'Cannot find module \'/Users/cyrussuen/RiTaJS/lib/rita\'' }
{ files: 0,
  assertions: 0,
  failed: 0,
  passed: 0,
  runtime: 0,
  tests: 0,
  coverage: 
   { files: 0,
     statements: { covered: 0, total: 0 },
     branches: { covered: 0, total: 0 },
     functions: { covered: 0, total: 0 },
     lines: { covered: 0, total: 0 } } }
[21:35:19] Finished 'test-only' after 3.42 s
```

```bash
Cyrus-Mac-Pro:RiTaJS cyrussuen$ gulp test-npm
[21:41:58] Using gulpfile ~/RiTaJS/gulpfile.js
[21:41:58] Starting 'clean'...
[21:41:58] Finished 'clean' after 79 ms
[21:41:58] Starting 'build-nolex'...
[21:41:58] Starting 'build-lex'...
[21:41:59] rita.js 147.59 kB
[21:41:59] Finished 'build-nolex' after 854 ms
[21:41:59] rita-full.js 1.86 MB
[21:41:59] Finished 'build-lex' after 867 ms
[21:41:59] Starting 'build'...
[21:41:59] Finished 'build' after 20 μs
[21:41:59] Starting 'test-npm'...
[21:41:59] Starting 'test-only'...
[INFO] Source: lib/rita.js
[21:42:04] Finished 'test-npm' after 5.07 s
{ [Error: Cannot find module '/Users/cyrussuen/RiTaJS/lib/rita']
  message: 'Cannot find module \'/Users/cyrussuen/RiTaJS/lib/rita\'' }
{ files: 0,
  assertions: 0,
  failed: 0,
  passed: 0,
  runtime: 0,
  tests: 0,
  coverage: 
   { files: 0,
     statements: { covered: 0, total: 0 },
     branches: { covered: 0, total: 0 },
     functions: { covered: 0, total: 0 },
     lines: { covered: 0, total: 0 } } }
[21:42:05] Finished 'test-only' after 5.7 s
```

**Comment by CyrusSUEN** [comment]
> asked you NOT to do it

Yes. I understand and never create the PR.
I made the commit just to verify the steps.

**Comment by dhowe** [comment]
ok, good... 
but aren't both 'npm test' and 'gulp test-npm'  failing?

**Comment by CyrusSUEN** [comment]
> but aren't both 'npm test' and 'gulp test-npm' failing?

Yes both failing.

**Comment by dhowe** [comment]
do they pass when you test the instructions from npm?

**Comment by CyrusSUEN** [comment]
> test the instructions from npm

I'm not sure what other instructions can be tested.
When running ```npm test``` from RiTaJS repo it prints the above error.

When running ```npm test``` from ```cdnjs/ajax/libs/rita/``` it prints "No gulpfile found"

**Comment by dhowe** [comment]
Didn't you test all the different instructions for the website?

<img width="648" alt="screen shot 2015-09-24 at 10 40 58 pm" src="https://cloud.githubusercontent.com/assets/737638/10076286/5b34bf36-630d-11e5-9e10-1153f7d25704.png">

**Comment by CyrusSUEN** [comment]
Yes. I did ```$ cd node_modules/rita && npm install && npm test``` after ```npm install rita```.

But inside RiTaJS repo, it cannot pass the test by running ```npm install && npm test``` with the above error.

**Comment by dhowe** [comment]
closing in favor of #340
